### PR TITLE
[6CORE-065] Create library for Unit Testing

### PR DESCRIPTION
**Pull request by andrey-kiyas**
- Created library structure
- Added all often-used testing dependency
- Consider using JUnit 5.9.X+
- Consider using Mockito 5.2.X+
- Registered library in 6core-BOM
- Added unit testing library to all app/domain modules

### CODE REVIEW

File: 2-library/unit-testing/pom.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.6core.lib</groupId>
+    <artifactId>unit-testing</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+<!--    <version>1.0-SNAPSHOT</version>-->
```
Comment (sokimaaa) [inline 2-library/unit-testing/pom.xml]: redundant comment
```suggestion
```

File: 2-library/unit-testing/pom.xml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.9.0</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>net.devh</groupId>
+            <artifactId>grpc-server-spring-boot-starter</artifactId>
+            <version>2.15.0.RELEASE</version>
+        </dependency>
```
Comment (sokimaaa) [inline 2-library/unit-testing/pom.xml]: This is not related to unit tests, isnt it?
```suggestion
```

File: 2-library/unit-testing/src/main/java/com/_6core/lib/unit/testing/UnitTestingAutoConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com._6core.lib.unit.testing;
+
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+
+@AutoConfiguration
```
Comment (sokimaaa) [inline 2-library/unit-testing/src/main/java/com/_6core/lib/unit/testing/UnitTestingAutoConfiguration.java]: we dont need to have spring here, just leave `package-info.java`

File: 2-library/unit-testing/pom.xml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    <properties>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
```
Comment (sokimaaa) [inline 2-library/unit-testing/pom.xml]: add unit testing library to all serivces (domain, application modules). 
Test that all works fine.

Comment (sokimaaa) [inline 2-library/unit-testing/pom.xml]: Do we need to have both? 
`junit-jupiter` and `junit-jupiter-api`?

junit-jupiter aggregates itself junit-jupiter-api and other depencency.

File: 2-library/unit-testing/pom.xml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <version>3.20.2</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.9.0</version>
```
Comment (sokimaaa) [inline 2-library/unit-testing/pom.xml]: pls add mockito-junit-jupiter
for using @ExtendWith(MockitoExtension.class) annotation

File: 2-library/unit-testing/pom.xml [hunk 6]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.6core.lib</groupId>
+    <artifactId>unit-testing</artifactId>
```
Comment (sokimaaa) [inline 2-library/unit-testing/pom.xml]: what do you think about such a name? 
e.g. library for bundling all depencencies needed for unit tests in our project
p.s. rename directory as well if agree 
```suggestion
    <artifactId>unit-test-bundle</artifactId>
```

**Review by sokimaaa — CHANGES_REQUESTED** [review]

File: 2-library/unit-testing/pom.xml (hunk 1 above)
Comment (andrey-kiyas) [reply 2-library/unit-testing/pom.xml]: Definitely redundant. I added the dependency because of an additional file that would help display the directory tree in the PR.

File: 2-library/unit-testing/pom.xml (hunk 2 above)
Comment (andrey-kiyas) [reply 2-library/unit-testing/pom.xml]: No, it isn't, 100% )

File: 2-library/unit-testing/src/main/java/com/_6core/lib/unit/testing/UnitTestingAutoConfiguration.java (hunk 3 above)
Comment (andrey-kiyas) [reply 2-library/unit-testing/src/main/java/com/_6core/lib/unit/testing/UnitTestingAutoConfiguration.java]: A pleasant innovation for me, it was possible to create a crutch file using native means, as it turned out ))

File: 2-library/unit-testing/pom.xml (hunk 4 above)
Comment (andrey-kiyas) [reply 2-library/unit-testing/pom.xml]: Yes, that's right, I left only `junit-jupiter`. THX.

File: 2-library/unit-testing/pom.xml (hunk 5 above)
Comment (andrey-kiyas) [reply 2-library/unit-testing/pom.xml]: I added `mockito-junit-jupiter` v5.11.0

File: 2-library/unit-testing/pom.xml (hunk 6 above)
Comment (andrey-kiyas) [reply 2-library/unit-testing/pom.xml]: Great. It's a pity it didn't occur to me myself. Thank you!

File: 2-library/6core-bom/pom.xml [hunk 7]
```diff
@@ -68,6 +69,12 @@
                 <version>0.0.1-SNAPSHOT</version>
             </dependency>
```
Comment (VictoriaKh-T) [inline 2-library/6core-bom/pom.xml]: You should "Registered library in 6core-BOM"

File: 3-platform/delivery/pom.xml [hunk 8]
```diff
@@ -21,4 +21,23 @@
     <module>spec</module>
   </modules>
 
+  <dependencies>
+    <dependency>
+      <groupId>com.6core.lib</groupId>
+      <artifactId>unit-test-bundle</artifactId>
+      <version>0.0.1-SNAPSHOT</version>
+    </dependency>
```
Comment (VictoriaKh-T) [inline 3-platform/delivery/pom.xml:28]: When you registered library in  "6core-BOM"  you can add it without version.

File: 2-library/unit-test-bundle/src/main/java/com/_6core/lib/unit/test/bundle/package-info.java [hunk 9]
```diff
@@ -0,0 +1,8 @@
+/**
+ * JUnit and Mockito have many dependencies that take up
+ * a lot of space and create a messy dependency block in pom.xml
+ * This module contains one library that will have everything
+ * these dependencies to just inject just one
+ */
```
Comment (sokimaaa) [inline 2-library/unit-test-bundle/src/main/java/com/_6core/lib/unit/test/bundle/package-info.java]: if you want to leave a comment in package-info you need to describe the needs of the package
this comment describes likely the whole library, not the package

for libs description you can use `README.md`
```suggestion
```

File: 3-platform/warehouse/pom.xml [hunk 10]
```diff
@@ -27,4 +27,24 @@
...
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>org.springframework.boot</groupId>
+          <artifactId>spring-boot-maven-plugin</artifactId>
+          <version>3.1.5</version>
+        </plugin>
+      </plugins>
+    </pluginManagement>
+  </build>
```
Comment (sokimaaa) [inline 3-platform/warehouse/pom.xml]: pls dont add code that is not related to the task directly. 
```suggestion
```

File: 3-platform/warehouse/pom.xml [hunk 11]
```diff
@@ -27,4 +27,24 @@
...
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
   </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.6core.lib</groupId>
+      <artifactId>unit-test-bundle</artifactId>
+      <version>0.0.1-SNAPSHOT</version>
+    </dependency>
+  </dependencies>
```
Comment (sokimaaa) [inline 3-platform/warehouse/pom.xml]: get rid of `version` -> we are using `6core-bom` for this. 
move dependency to the app/domain module. 
replace all `junit`, `mockito` dependencies with your unit test bundle.
```suggestion
  <dependencies>
    <dependency>
      <groupId>com.6core.lib</groupId>
      <artifactId>unit-test-bundle</artifactId>
    </dependency>
  </dependencies>
```

File: 2-library/unit-test-bundle/pom.xml [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>5.2.0</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-junit-jupiter</artifactId>
+            <version>5.11.0</version>
```
Comment (sokimaaa) [inline 2-library/unit-test-bundle/pom.xml]: use the same version for all dependencies to avoid conflict. 
*that related to one group, e.g. `org.mockito`
```suggestion
            <version>5.2.0</version>
```

**Review by sokimaaa — CHANGES_REQUESTED** [review]

File: 3-platform/warehouse/pom.xml (hunk 10 above)
Comment (sokimaaa) [reply 3-platform/warehouse/pom.xml]: if you need this for some reason, pls move to the BOM

File: 2-library/unit-test-bundle/src/main/java/com/_6core/lib/unit/test/bundle/package-info.java (hunk 9 above)
Comment (andrey-kiyas) [reply 2-library/unit-test-bundle/src/main/java/com/_6core/lib/unit/test/bundle/package-info.java]: The idea was not well thought out. Just deleted it. )

File: 3-platform/warehouse/pom.xml (hunk 10 above)
Comment (andrey-kiyas) [reply 3-platform/warehouse/pom.xml]: I worked on the issue more deeply. The plugin did not have enough version to work properly. (at order/boot/pom.xml)

File: 2-library/unit-test-bundle/pom.xml (hunk 12 above)
Comment (andrey-kiyas) [reply 2-library/unit-test-bundle/pom.xml]: I agree, it's done

File: 3-platform/warehouse/pom.xml (hunk 11 above)
Comment (andrey-kiyas) [reply 3-platform/warehouse/pom.xml]: Yes, this option is much better and simpler, thx

**Comment by andrey-kiyas** [comment]
image
This mean I can't make changes to the BOM

**Comment by andrey-kiyas** [comment]
Error:  'dependencies.dependency.version' for com.6core.lib:unit-test-bundle:jar is missing. @ line 20, column 17

I get the expected error - I don’t know how to get past it without making changes to the `6core-bom`

**Review by sokimaaa — APPROVED** [review]

**Review by VictoriaKh-T — APPROVED** [review]
